java/arcs/android/storage/service/StorageServiceNgImpl.kt

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2020 Google LLC.
+ *
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ *
+ * Code distributed by Google as part of this project is also subject to an additional IP rights
+ * grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+package arcs.android.storage.service
+
+import android.os.Parcel
+import androidx.annotation.VisibleForTesting
+import arcs.android.storage.readStoreOptions
+import arcs.core.storage.ActiveStore
+import arcs.core.storage.DriverFactory
+import arcs.core.storage.StorageKey
+import arcs.core.storage.UntypedActiveStore
+import arcs.core.storage.UntypedProxyMessage
+import arcs.core.storage.WriteBackProvider
+import java.util.concurrent.ConcurrentHashMap
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.launch
+
+/**
+ * Implementation of the [IStorageServiceNg] AIDL interface. Responsible for forwarding messages
+ * to [ActiveStore]s and back again.
+ */
+class StorageServiceNgImpl(
+  private val scope: CoroutineScope,
+  private val driverFactory: DriverFactory,
+  private val writeBackProvider: WriteBackProvider,
+  private val devToolsProxy: DevToolsProxyImpl?,
+  /** Callback to trigger when a proxy message has been received and sent to the store. */
+  private val onProxyMessageCallback: suspend (StorageKey, UntypedProxyMessage) -> Unit
+) : IStorageServiceNg.Stub() {
+  // TODO(b/162747024): Replace this with an LruCache so its size doesn't grow unbounded.
```

Comment:
I don't think that using an `LruCache` here is what we want. We should be able to control the growth of this map of stores by making sure to clean it up when all endpoints using it are closed.

Reply:
In fact, I'm almost certain that we *definitely* don't want an LruCache here. It would lead to strange behavior at the size limit. For example, handles may stop getting update messages because the underlying store has disappeared.

Reply:
Ok I've removed that TODO and added a different TODO to "Clean up stores when the channels. are all closed/dead".